tutorials/object_detection.ipynb

```diff
@@ -0,0 +1,269 @@
+       "version_minor": 0
+      },
+      "text/plain": [
+       "Video(value=b'\\x00\\x00\\x00 ftypisom\\x00\\x00\\x02\\x00isomiso2avc1mp41\\x00\\x00\\x00\\x08free\\x00\\x19K[mdat\\x00\\x00\\…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "from ipywidgets import Video\n",
+    "Video.from_file(\"../data/ua_detrac/ua_detrac.mp4\", embed=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "686adeb5",
+   "metadata": {},
+   "source": [
+    "### Load video into EVA"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "b18cfe2a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cursor.execute('LOAD DATA INFILE \"test_video.mp4\" INTO MyVideo;')\n",
+    "response = cursor.fetch_one()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "740b38cb",
+   "metadata": {},
+   "source": [
+    "### Optional - Register FasterRCNN (object detection) model into EVA "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "e83e5a44",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cursor.execute(\"\"\"CREATE UDF IF NOT EXISTS FastRCNNObjectDetector \n",
+    "                  INPUT  (frame NDARRAY UINT8(3, 256, 256))\n",
+    "                  OUTPUT (labels NDARRAY STR(20), bboxes NDARRAY INT32(20, 4))\n",
+    "                  TYPE  Classification\n",
+    "                  IMPL  'src/udfs/fastrcnn_object_detector.py'\"\"\")\n",
+    "response = cursor.fetch_one()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7bbd789e",
+   "metadata": {},
+   "source": [
+    "### Run Object detector on the video"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "91bdcaca",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cursor.execute(\"\"\"SELECT id, Unnest(FastRCNNObjectDetector(data)) FROM MyVideo\"\"\")\n",
+    "response = cursor.fetch_one()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d81ed233",
+   "metadata": {},
+   "source": [
+    "### Visualize output of Object detector on the video"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "ecc977d8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cv2\n",
+    "def annotate_video(input_path, output_path):\n",
+    "    color=(0,255,0)\n",
+    "    thickness=3\n",
+    "\n",
+    "    vcap = cv2.VideoCapture(input_path)\n",
+    "    width = int(vcap.get(3))\n",
+    "    height = int(vcap.get(4))\n",
+    "    fps = vcap.get(5)\n",
+    "    fourcc = cv2.VideoWriter_fourcc(*'H264') #codec\n",
+    "    video=cv2.VideoWriter(output_path, fourcc, fps, (width,height))\n",
+    "\n",
+    "    frame_id = 0\n",
+    "    # Capture frame-by-frame\n",
+    "    ret, frame = vcap.read()  # ret = 1 if the video is captured; frame is the image\n",
+    "\n",
+    "    while ret:\n",
+    "        df = response.batch.frames\n",
```

Comment:
Might be better to add the variable `response` as a function argument instead of directly accessing the global variable.

Reply:
Yes, correct. I'll modularize it.